Packages/com.unity.inputsystem/InputSystem/Plugins/EnhancedTouch/TouchSimulation.cs

```diff
@@ -264,6 +273,28 @@ protected void OnEnable()
 
             InputSystem.onDeviceChange += m_OnDeviceChange;
             InputSystem.onEvent += m_OnEvent;
+            
+            // In case there's a PlayerInput component in the scene, we want to pair the simulated touchscreen to
+            // the PlayerInput user. The touchscreen device does not queue events so we need to pair it to the user.
+            m_PlayerInput = PlayerInput.s_AllActivePlayersCount > 0 ? PlayerInput.s_AllActivePlayers[0] : null; 
```

Comment:
I find it interesting in general that the existing TouchSimulation is modifying state memory instead of acting as a proper touch device and enqueuing events. Did you by any chance see anything in implementation why that would be the case?

Reply:
I think a follow-up task to make TouchSimulated to use input events could be added to the backlog as it would make it much more "real" than it currently is. I'll deal with that.